### PR TITLE
add patch to increase compatibility with AVX512 platforms for bwa-mem2 v2.2.1

### PR DESCRIPTION
**Pull request by lexming**

### CODE REVIEW

File: easybuild/easyconfigs/b/bwa-mem2/bwa-mem2-2.2.1-intel-compilers-2023.1.0.eb [hunk 1]
```diff
@@ -15,10 +15,14 @@ toolchainopts = {'pic': True, 'oneapi': False}
 github_account = 'bwa-mem2'
 source_urls = [GITHUB_SOURCE]
 sources = ['v%(version)s.tar.gz']
-patches = ['%(name)s-%(version)s_use_external_deps.patch']
+patches = [
+    'bwa-mem2-%(version)s_use_external_deps.patch',
+    'bwa-mem2-%(version)s_use_common_avx512flags.patch',
```
Comment (verdurin) [inline easybuild/easyconfigs/b/bwa-mem2/bwa-mem2-2.2.1-intel-compilers-2023.1.0.eb:20]:
```suggestion
    'bwa-mem2-%(version)s_common_avx512flags.patch',
```
Or rename the patch to be consistent with the reference here.

**Review by verdurin — CHANGES_REQUESTED** [review]

File: easybuild/easyconfigs/b/bwa-mem2/bwa-mem2-2.2.1-intel-compilers-2023.1.0.eb (hunk 1 above)
Comment (lexming) [reply easybuild/easyconfigs/b/bwa-mem2/bwa-mem2-2.2.1-intel-compilers-2023.1.0.eb:20]: @verdurin thanks! I'm dumb :smile:

**Comment by lexming** [comment]
@boegelbot: please test @ generoso

**Comment by lexming** [comment]
@boegelbot please test @ jsc-zen3

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node600.hydra.os - Linux Rocky Linux 8.9, x86_64, AMD EPYC 9384X 32-Core Processor (x86_64_v4), Python 3.6.8
See https://gist.github.com/lexming/4967cd0e41fbd28eee54c58ebe38e188 for a full test report.

**Comment by lexming** [comment]
Test report by @lexming
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node300.hydra.os - Linux Rocky Linux 8.9, x86_64, Intel(R) Xeon(R) Gold 6148 CPU @ 2.40GHz (skylake_avx512), Python 3.6.8
See https://gist.github.com/lexming/af8a6262bcaa3febd074027969fd7e11 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 2 out of 2 (1 easyconfigs in total)
node3117.skitty.os - Linux RHEL 8.8, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/boegel/1c05ba7df1d0221d4a7e2a8dda13ef39 for a full test report.

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Going in, thanks @lexming!

**Pull request merged**